processor/marshalprocessor/README.md

```diff
@@ -0,0 +1,192 @@
+# Marshal Processor
+
+This processor is used to marshal parsed logs into JSON or KV format.
+
+This processor is intended to be wrapped into the Marshal processor in Bindplane.
+
+NOTE: XML support is in progress and not yet available.
+
+## Supported pipelines
+
+- Logs
+
+## How It Works
+
+1. This processor expects its input to contain a parsed log body. It will marshal the body fields only; if additional fields from the log are desired, they must first be moved to the body.
+
+2. The body can be marshaled to string-encoded JSON or KV.
+
+   - For KV:
+     - Fields will be converted to "key1=value1 key2=value2 key3=value3..." if no separators are configured
+     - The parsed fields should be flattened first so that every key is at the top level
+     - If fields are not flattened, the nested fields will be converted to "nested=[k1=v1,k2=v2]..." if no map separators are configured
+     - If any key or value contains characters that conflict with the separators, they will be wrapped in `"` and any `"` inside them will be escaped
+     - Arrays will simply be stringified
+
+3. The output of this processor will be the same as the input, but with a modified log body. Any body incompatible with the marshal type will be unchanged.
+
+## Configuration
+
+| Field              | Type   | Default | Description                                   |
+| ------------------ | ------ | ------- | --------------------------------------------- |
+| marshalTo          | string | ""      | The format to marshal into. Can be JSON or KV |
+| kvSeparator        | rune   | "="     | The separator between key and value           |
+| kvPairSeparator    | rune   | " "     | The separator between KV pairs                |
+| mapKVSeparator     | rune   | "="     | The separator between nested KV pairs         |
+| mapKVPairSeparator | rune   | ","     | The separator between nested KV pairs         |
```

Comment:
Testing this out, looks like these field names should be in `snake_case`

Reply:
This also applies to the config samples below (e.g. marshalTo should actually be `marshal_to`)